### PR TITLE
Lucho plan 1783 planning area for treatments

### PR DESCRIPTION
**Pull request by LALuis**
The idea of this PR is to add the planning area for the treatments view.

To achieve that we created a new component that gets the plan based on the plan id extracted from the url and draws the area based on the geometry of that plan.

Jira: https://sig-gis.atlassian.net/browse/PLAN-1783

Result:
image

### CODE REVIEW

File: src/interface/src/app/treatments/treatment-area-layer/treatment-area-layer.component.html [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+    'line-color': 'black',
+    'line-width': 2
+  }">
+</mgl-layer>
+<mgl-layer
+  id="treatment-area-fill"
+  type="fill"
+  [source]="sourceName"
+  [paint]="{
+    'fill-color': '#e9eded',
```
Comment (LALuis) [inline src/interface/src/app/treatments/treatment-area-layer/treatment-area-layer.component.html:18]: Let me know if this color is ok, I picked it up from the area one using the devtools but I don't know if its clear enough

File: src/interface/src/app/treatments/treatment-area-layer/treatment-area-layer.component.ts [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+import { GeoJSONSource, Map as MapLibreMap } from 'maplibre-gl';
+
+@Component({
+  selector: 'app-treatment-area-layer',
+  standalone: true,
+  imports: [FeatureComponent, GeoJSONSourceComponent, LayerComponent],
+  templateUrl: './treatment-area-layer.component.html',
+  styleUrl: './treatment-area-layer.component.scss',
+})
+export class TreatmentAreaLayerComponent implements OnInit {
```
Comment (pflopez) [inline src/interface/src/app/treatments/treatment-area-layer/treatment-area-layer.component.ts:19]: Can you rename this to PlanningAreaLayer ? This layer is showing the planning area

File: src/interface/src/app/treatments/treatment-area-layer/treatment-area-layer.component.ts [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+  readonly sourceName = 'tratment-planing-area';
+
+  polygonGeometry: Geometry = {
+    type: 'Polygon',
+    coordinates: [[]],
+  };
+
+  ngOnInit(): void {
+    this.route.paramMap.subscribe((params) => {
+      this.planId = params.get('planId') || '';
```
Comment (pflopez) [inline src/interface/src/app/treatments/treatment-area-layer/treatment-area-layer.component.ts:36]: Question, maybe the `planId` can be an @Input instead? so we don't couple routing stuff on this component, and keep it all on the parent one? What do you think? Might simplify this one a bit.
Another alternative, we are already setting some of the route info in TreatmentsState, maybe we should add the planId there? Like maybe update TreatmentsState.setInitialState to also store the planId ?

File: src/interface/src/app/treatments/treatment-area-layer/treatment-area-layer.component.ts (hunk 2 above)
Comment (LALuis) [reply src/interface/src/app/treatments/treatment-area-layer/treatment-area-layer.component.ts:19]: Updating.....

File: src/interface/src/app/treatments/treatment-area-layer/treatment-area-layer.component.ts (hunk 3 above)
Comment (LALuis) [reply src/interface/src/app/treatments/treatment-area-layer/treatment-area-layer.component.ts:36]: That is a good idea, let me take a look <3

File: src/interface/src/app/treatments/treatments.state.ts [hunk 4]
```diff
@@ -46,6 +46,8 @@ export class TreatmentsState {
   private _projectAreaId$ = new BehaviorSubject<number | undefined>(undefined);
   public projectAreaId$ = this._projectAreaId$.asObservable();
 
+  public planId$ = new BehaviorSubject<any>(null);
```
Comment (pflopez) [inline src/interface/src/app/treatments/treatments.state.ts:49]: types here please :)

File: src/interface/src/app/treatments/treatments-routing-data.ts [hunk 5]
```diff
@@ -5,6 +5,7 @@ export interface TreatmentRoutingData extends Data {
   scenarioId: number;
   treatmentId: number;
   projectAreaId?: number;
+  planId?: number;
```
Comment (pflopez) [inline src/interface/src/app/treatments/treatments-routing-data.ts:8]: lets make this required- we should always have planId given our routing structure

Comment (LALuis) [reply src/interface/src/app/treatments/treatments-routing-data.ts:8]: Updated, you will see it in the next commit

File: src/interface/src/app/treatments/treatments.state.ts (hunk 4 above)
Comment (LALuis) [reply src/interface/src/app/treatments/treatments.state.ts:49]: Ohh sorry! You will see that in my next commit

File: src/interface/src/app/treatments/treatment-map/treatment-map.component.ts [hunk 6]
```diff
@@ -142,10 +146,21 @@ export class TreatmentMapComponent {
    */
   treatmentTooltipLngLat: LngLat | null = null;
 
+  /**
+   *
+   * The selected plan geometry
+   */
+  geometry: Geometry = {
```
[the remark this answers is not in the document]
Comment (LALuis) [reply src/interface/src/app/treatments/treatment-map/treatment-map.component.ts]: You already did it :D

File: src/interface/src/app/treatments/treatments-routing-data.ts [hunk 7]
```diff
@@ -5,6 +5,7 @@ export interface TreatmentRoutingData extends Data {
   scenarioId: number;
   treatmentId: number;
   projectAreaId?: number;
+  planId: number | null;
```
[the remark this answers is not in the document]
Comment (LALuis) [reply src/interface/src/app/treatments/treatments-routing-data.ts]: You are right :D updated

**Review by pflopez — APPROVED** [review]

**Pull request merged**